app/grandchallenge/utilization/models.py

```diff
@@ -99,3 +100,101 @@ class Meta:
                 name="unique_session_utilization_reader_study",
             )
         ]
+
+
+class ComponentJobUtilizationManager(models.QuerySet):
+    def average_duration(self):
+        """Calculate the average duration that completed jobs ran for"""
+        return self.exclude(duration=None).aggregate(
+            duration__avg=Avg("duration")
+        )["duration__avg"]
+
+
+class ComponentJobUtilization(UUIDModel):
+    creator = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="%(class)ss",
+    )
+    phase = models.ForeignKey(
+        "evaluation.Phase",
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="%(class)ss",
+    )
+    challenge = models.ForeignKey(
+        "challenges.Challenge",
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="%(class)ss",
+    )
+    archive = models.ForeignKey(
+        "archives.Archive",
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="%(class)ss",
+    )
+    algorithm_image = models.ForeignKey(
+        "algorithms.AlgorithmImage",
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="%(class)ss",
+    )
+    algorithm = models.ForeignKey(
+        "algorithms.Algorithm",
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="%(class)ss",
+    )
+    duration = models.DurationField(null=True)
+    compute_cost_euro_millicents = models.PositiveIntegerField(null=True)
+
+    objects = ComponentJobUtilizationManager.as_manager()
+
+    class Meta:
+        abstract = True
+
+
+class JobUtilization(ComponentJobUtilization):
+    job = models.OneToOneField(
+        "algorithms.Job",
+        related_name="jobutilization",
```

Comment:
`jobutilization` and `evaluationutilization` is really ugly, where did the underscore go? Adding them back would be better, or even using `utilization` for both.

Reply:
I think that you could set:

```python3
class Meta(ComponentJobUtilization.Meta):
  default_related_name = "job_utilizations"
```

etc. in each of the concrete classes and leave them unset in the fields. See https://docs.djangoproject.com/en/5.2/ref/models/options/